demos/object_detection_demo_ssd_async/main.cpp

```diff
@@ -235,102 +292,92 @@ int main(int argc, char *argv[]) {
                 data[1] = static_cast<float>(netInputWidth);  // width
                 data[2] = 1;
             };
-            setImgInfoBlob(async_infer_request_curr);
-            setImgInfoBlob(async_infer_request_next);
+
+            for (const InferRequest::Ptr &requestPtr : userSpecifiedInferRequests) {
+                setImgInfoBlob(requestPtr);
+            }
+            setImgInfoBlob(minLatencyInferRequest);
         }
         // -----------------------------------------------------------------------------------------------------
 
-        // --------------------------- 6. Do inference ---------------------------------------------------------
-        slog::info << "Start inference " << slog::endl;
-
-        bool isLastFrame = false;
-        bool isAsyncMode = false;  // execution is always started using SYNC mode
-        bool isModeChanged = false;  // set to TRUE when execution mode is changed (SYNC<->ASYNC)
+        // --------------------------- 6. Init variables -------------------------------------------------------
+        typedef std::chrono::duration<double, std::chrono::milliseconds::period> Ms;
+        typedef std::chrono::duration<double, std::chrono::seconds::period> Sec;
+
+        struct RequestResult {
+            cv::Mat frame;
+            Blob::Ptr output;
+            std::chrono::steady_clock::time_point startTime;
+            bool isSameMode;
+        };
+
+        struct ModeInfo {
+           int framesCount = 0;
+           std::chrono::steady_clock::duration latencySum;
+           std::chrono::steady_clock::time_point lastStartTime;
+           std::chrono::steady_clock::time_point lastEndTime;
+        };
+
+        auto totalT0 = std::chrono::steady_clock::now();
+
+        // execution always starts in USER_SPECIFIED mode
+        enum class ExecutionMode {userSpecified, minLatency};
```

Comment:
```suggestion
        enum class ExecutionMode {USER_SPECIFIED, MIN_LATENCY};
```

Not only is this the convention for enum constants, but you even wrote `USER_SPECIFIED` yourself above. :)